packages/metadataform/src/metadataform.ts

```diff
@@ -0,0 +1,473 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+/**
+ * @packageDocumentation
+ * @module metadataform
+ */
+
+import { NotebookTools } from '@jupyterlab/notebook';
+import { IObservableJSON } from '@jupyterlab/observables';
+import {
+  ITranslator,
+  nullTranslator,
+  TranslationBundle
+} from '@jupyterlab/translation';
+import {
+  JSONExt,
+  PartialJSONObject,
+  PartialJSONValue,
+  ReadonlyPartialJSONObject,
+  ReadonlyPartialJSONValue
+} from '@lumino/coreutils';
+import { Message } from '@lumino/messaging';
+import { SingletonLayout, Widget } from '@lumino/widgets';
+
+import { MetadataForm } from './token';
+import { FormWidget } from './form';
+
+/**
+ * A class that create a metadata form widget
+ */
+export class MetadataFormWidget
+  extends NotebookTools.Tool
+  implements MetadataForm.IMetadataForm
+{
+  /**
+   * Construct an empty widget.
+   */
+  constructor(
+    metadataSchema: MetadataForm.IMetadataSchema,
+    metaInformation: MetadataForm.IMetaInformation,
+    uiSchema: MetadataForm.IUiSchema,
+    pluginId?: string,
+    translator?: ITranslator,
+    showModified?: boolean
```

Comment:
In the code base, we tend to use an object `options` rather than long list of arguments to be able to easily extend it and skip non-mandatory argument.

> This is a general comment, I'm fine with keeping things as is.

Reply:
Right thanks, I'll change it.

Reply:
Done in https://github.com/jupyterlab/jupyterlab/pull/13056/commits/77b9bd39ea421b43fbba692eed8f942124d3a695